scripts/check_if_pr_is_low_risk.py

```diff
@@ -0,0 +1,317 @@
+
+import argparse
+import json
+import re
+import sys
+
+import python_utils
+from scripts import common
+
+
+GITHUB_API_PR_ENDPOINT = (
+    'https://api.github.com/repos/%s/%s/pulls/%s')
+PR_URL_REGEX = (
+    r'^https://github.com/(?P<owner>\w+)/(?P<repo>\w+)/pull/(?P<num>\w+)/?$')
+UPSTREAM_REMOTE = 'upstream'
+FILES_THAT_NEED_DIFFS = (
+    'package.json'
+    'core/templates/pages/about-page/about-page.constants.ts',
+)
+
+
+def parse_pr_url(pr_url):
+    """Extract the owner, repo, and PR number from a PR URL.
+
+    For example, in the URL https://github.com/foobar/oppia/pull/23, the
+    owner is `foobar`, the repo is `oppia`, and the PR number is `23`.
+
+    Args:
+        pr_url: str. URL to the PR on GitHub.
+
+    Returns:
+        str, str, str. 3-tuple of the PR owner, repository, and number.
+    """
+    match = re.match(PR_URL_REGEX, pr_url)
+    if match:
+        return match.group('owner', 'repo', 'num')
+    else:
+        return None
+
+
+def load_diff(base_branch):
+    """Load the diff between the head and base.
+
+    Only determine the diffs for files in FILES_THAT_NEED_DIFFS. Other
+    files will be listed as having changed, but their diff lines will
+    not be returned.
+
+    Args:
+        base_branch: str. Base branch of PR.
+
+    Returns:
+        tuple(list(tuple(str, str)), dict(str, list(str)). Tuple of a
+        list of changed files (each a tuple of before, after) and a
+        dictionary mapping from file names to list of diff lines for
+        each file. In the event of a parsing error, a tuple of an empty
+        list and empty dictionary is returned.
+    """
+    diff_name_status = common.run_cmd([
+        'git', 'diff', '--name-status',
+        '{}/{}'.format(UPSTREAM_REMOTE, base_branch),
+    ])
+    diff_files = []
+    for line in diff_name_status.split('\n'):
+        if not line:
+            continue
+        split = line.split()
+        if len(split) < 2 or len(split) > 3:
+            python_utils.PRINT(
+                'Failed to parse diff --name-status line "%s"'
+                % line)
+            return [], {}
+        elif len(split) == 2:
+            diff_files.append((split[1], split[1]))
+        elif len(split) == 3:
+            diff_files.append((split[1], split[2]))
+    file_diffs = {}
+    for file_tuple in diff_files:
+        for filename in file_tuple:
+            if filename in file_diffs:
+                continue
+            if filename not in FILES_THAT_NEED_DIFFS:
+                continue
+            file_diff = common.run_cmd([
+                'git', 'diff', '-U0',
+                '{}/{}'.format(UPSTREAM_REMOTE, base_branch),
+                '--', filename,
+            ])
+            file_diff_split = file_diff.rstrip().split('\n')
+            i = 0
+            for line in file_diff_split:
+                i += 1
+                if line.startswith('@@'):
+                    break
+            if i > 6 or i == len(file_diff_split):
```

Comment:
Diffs have this form:

```
diff --git a/start_file b/end_file
index start_commit..end_commit ...
--- a/start_file
+++ b/end_file
@@ <line numbers>
```

We should never see more than 5 lines before the first `@@`. Since `i` is the index of the line after the first `@@`, `i` should never be above 6.

Reply:
Actually, I'm removing the `6` part of this check since diff headers can be arbitrarily long: https://git-scm.com/docs/diff-format